lcm-logger/lcm_logger.c

```diff
@@ -276,7 +452,26 @@ static void *write_thread(void *user_data)
             logger->events_since_last_report = 0;
             logger->last_report_logsize = logger->logsize;
         }
-    }
+
+        // ---
+        // Disk Quota @Review: This whole section
+
+        // judfs: Should this be its own thread?
+        // Should the interval be a flag?
+        const int64_t quota_check_interval = 1e6;
+        if (logger->disk_quota && (offset_utime - logger->last_quota_time > quota_check_interval)) {
+            logger->last_quota_time = offset_utime;
+
+            int64_t free_bytes = get_free_disk_bytes(logger->write_directory);
+            if (free_bytes < logger->disk_quota) {
+                printf("Disk quota exceeded. Exiting logger.\n");
+                // @Review: How should this exit? For now going with a graceful exit.
+                // But maybe this case warrants more urgency.
+                g_main_loop_quit(_mainloop);
```

Comment:
I think the current approach is ideal. If the user sets up a condition for the logger to end then a non-zero exit code makes sense to me, since everything is operating as expected. But I don't feel strongly at all about it.